griptape/drivers/ruleset/griptape_cloud_ruleset_driver.py

```diff
@@ -0,0 +1,90 @@
+from __future__ import annotations
+
+import os
+from typing import TYPE_CHECKING, Any, Optional
+from urllib.parse import urljoin
+
+import requests
+from attrs import Attribute, Factory, define, field
+
+from griptape.drivers import BaseRulesetDriver
+from griptape.utils import dict_merge
+
+if TYPE_CHECKING:
+    from griptape.rules import BaseRule
+
+
+@define(kw_only=True)
+class GriptapeCloudRulesetDriver(BaseRulesetDriver):
+    """A driver for storing conversation memory in the Griptape Cloud.
+
+    Attributes:
+        ruleset_id: The ID of the Thread to store the conversation memory in. If not provided, the driver will attempt to
+            retrieve the ID from the environment variable `GT_CLOUD_THREAD_ID`. If that is not set, a new Thread will be
+            created.
+        base_url: The base URL of the Griptape Cloud API. Defaults to the value of the environment variable
+            `GT_CLOUD_BASE_URL` or `https://cloud.griptape.ai`.
+        api_key: The API key to use for authenticating with the Griptape Cloud API. If not provided, the driver will
+            attempt to retrieve the API key from the environment variable `GT_CLOUD_API_KEY`.
+
+    Raises:
+        ValueError: If `api_key` is not provided.
+    """
+
+    ruleset_id: str = field(
+        default=None,
+        metadata={"serializable": True},
+    )
+    base_url: str = field(
+        default=Factory(lambda: os.getenv("GT_CLOUD_BASE_URL", "https://cloud.griptape.ai")),
```

Comment:
We should apply this pattern to all of our GriptapeCloud drivers (GT_Cloud mixin?). Looks like some use GT_CLOUD_BASE_URL, some just default to cloud.griptape.ai (Not this PR)

Reply:
this issue goes away with a proper SDK